hdk/jack/syntax.py

```diff
@@ -0,0 +1,614 @@
+            ")",
+            self.body,
+        )
+        return element
+
+    def __post_init__(self):
+        if not _is_symbol_valid(self.type_):
+            raise ValueError(f"Invalid type {self.type_!r} for subroutine declaration.")
+        if not _is_symbol_valid(self.returns):
+            raise ValueError(
+                f"Invalid returns {self.returns!r} for subroutine declaration."
+            )
+        if not _is_symbol_valid(self.name):
+            raise ValueError(f"Invalid name {self.name!r} for subroutine declaration.")
+
+
+@dataclass(frozen=True)
+class ClassVarDeclaration(AbstractSyntaxTree):
+    """Represents a class variable declaration.
+
+    Attributes:
+        modifier: The modifier of the variable declaration.
+        type_: The type of the variable.
+        names: The names of the variables.
+    """
+
+    modifier: str
+    type_: str
+    names: list[str]
+
+    @property
+    def is_identifier(self) -> bool:
+        """True if type is identifier."""
+        return self.type_ not in {"int", "char", "boolean"}
+
+    def to_xml(self, doc: Document) -> Element:
+        element = doc.createElement("classVarDec")
+        tag = "identifier" if self.is_identifier else "keyword"
+        _add_children(element, ("keyword", self.modifier), (tag, self.type_))
+        for i, name in enumerate(self.names):
+            _add_child(element, name, "identifier")
+            if i < len(self.names) - 1:
+                _add_child(element, ",")
+        _add_child(element, ";")
+        return element
+
+    def __post_init__(self):
+        if not _is_symbol_valid(self.type_):
+            raise ValueError(f"Invalid type {self.type_!r} for class var declaration.")
+        if not _is_symbol_valid(self.modifier):
+            raise ValueError(
+                f"Invalid modifier {self.modifier!r} for class var declaration."
+            )
+        for name in self.names:
+            if not _is_symbol_valid(name):
+                raise ValueError(f"Invalid name {name!r} for class var declaration.")
+
+
+@dataclass(frozen=True)
+class Class(AbstractSyntaxTree):
+    """Represents a class.
+
+    Attributes:
+        name: The name of the class.
+        class_vars: The class variable declarations.
+        subroutines: The subroutine declarations.
+    """
+
+    name: str
+    class_vars: list[ClassVarDeclaration]
+    subroutines: list[SubroutineDeclaration]
+
+    def to_xml(self, doc: Document) -> Element:
+        element = doc.createElement("class")
+        _add_children(
+            element,
+            ("keyword", "class"),
+            ("identifier", self.name),
+            "{",
+            self.class_vars,
+            self.subroutines,
+            "}",
+        )
+        return element
+
+    def __post_init__(self):
+        if not _is_symbol_valid(self.name):
+            raise ValueError(f"Invalid name {self.name!r} for class.")
+
+
+def _is_symbol_valid(symbol: str) -> bool:
```

Comment:
_is_symbol_valid -> _is_identifier_valid.
Also, allowed identifier check is different from valid symbol check. Find valid identifier description in the book. Also check that identifier could be one of keywords. Fix docstring accordingly.